enhancements/hypershift/disaster-recovery/hcp-etcd-backup-api.md

```diff
@@ -0,0 +1,587 @@
+- `hostedcluster.v1beta1.hypershift.openshift.io.spec.etcd.backup.etcdbackup.retentionPolicy`
+- `hostedcluster.v1beta1.hypershift.openshift.io.spec.etcd.backup.etcdbackup.retentionPolicy.retentionType`
+- `hostedcluster.v1beta1.hypershift.openshift.io.spec.etcd.backup.etcdbackup.retentionPolicy.retentionNumber`
+- `hostedcluster.v1beta1.hypershift.openshift.io.spec.etcd.backup.etcdbackup.retentionPolicy.retentionNumber.maxNumberOfBackups`
+- `hostedcluster.v1beta1.hypershift.openshift.io.spec.etcd.backup.etcdbackup.retentionPolicy.retentionSize`
+- `hostedcluster.v1beta1.hypershift.openshift.io.spec.etcd.backup.etcdbackup.retentionPolicy.retentionSize.maxSizeOfBackupsGb`
+- `hostedcluster.v1beta1.hypershift.openshift.io.spec.etcd.backup.storeEndpoint.backupPrefix`
+- `hostedcluster.v1beta1.hypershift.openshift.io.spec.etcd.backup.storeEndpoint.credentials`
+- `hostedcluster.v1beta1.hypershift.openshift.io.spec.etcd.backup.storeEndpoint.serviceConfig`
+- `hostedcluster.v1beta1.hypershift.openshift.io.spec.etcd.backup.storeEndpoint.serviceConfig.serviceURL`
+- `hostedcluster.v1beta1.hypershift.openshift.io.spec.etcd.backup.storeEndpoint.serviceConfig.region`
+- `hostedcluster.v1beta1.hypershift.openshift.io.spec.etcd.backup.storeEndpoint.serviceConfig.tags`
+
+Main security points:
+
+- Same secret management as being carried out by the HO, CPO, and HCCO.
+- No additional permissions needed to perform and manage the backups.
+- We will segregate permissions to create the secret in the HostedCluster namespace and replicate those secrets to the HostedControlPlane namespace by the controller.
+- The [persona](https://hypershift-docs.netlify.app/reference/concepts-and-personas/) that will benefit from this new API will be the Cluster Service Provider and the Cluster Service Consumer.
+- Like other endpoints in Hypershift, the inputs are restricted by the defined types:
+  - Enums are limited to the types defined in the code.
+  - Strings are restricted to a defined number of characters via OpenAPI.
+  - Integers are defined by the Go language.
+- All mentioned endpoints require authentication.
+- The backup performed is encrypted by default using the key generated for that ETCD cluster.
+- The permissions of the provided credentials could be very limited, only allowing for pushing the backup binary to a specific S3 bucket. Listing, modifying, or similar actions are not necessary.
+- This new component has not yet undergone DUST testing or penetration testing.
+
+**How will security be reviewed and by whom?**
+
+From RH Security team
+
+- David Sastre (@dawud) (Security Architect dedicated to Hypershift)
+- Florencio Cano (Backup)
+
+From the internal team:
+
+- @derekwaynecarr
+- @csrwng
+- @enxebre
+
+**How will UX be reviewed and by whom?**
+
+- @hypershift-team
+
+### Drawbacks
+
+N/A
+
+## Open Questions
+
+- Do we need a retryPolicy for the failure processes?
+- The Garbage collector will also clean the remote backups?
```

Comment:
That might not be necessary. object stores like s3 usually have retention policies.

Reply:
Makes sense. I will keep this on hold until a customer would needed at some point :). Thanks for the feedback.